tests_e2e/tests/ext_telemetry_pipeline/ext_telemetry_pipeline.py

```diff
@@ -0,0 +1,111 @@
+#  (work item https://dev.azure.com/msazure/One/_workitems/edit/24903999)
+#
+
+import random
+from typing import List, Dict, Any
+
+from azurelinuxagent.common.conf import get_etp_collection_period
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.identifiers import VmExtensionIds
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+
+
+class ExtTelemetryPipeline(AgentTest):
+    def run(self):
+        ssh_client: SshClient = self._context.create_ssh_client()
+
+        # Extensions we will create events for
+        extensions = ["Microsoft.Azure.Extensions.CustomScript"]
+        if VmExtensionIds.VmAccess.supports_distro(ssh_client.run_command("uname -a")):
+            extensions.append("Microsoft.OSTCExtensions.VMAccessForLinux")
+
+        # Set the etp collection period to 30 seconds instead of default 5 minutes
+        default_collection_period = get_etp_collection_period()
+        log.info("")
+        log.info("Set ETP collection period to 30 seconds on the test VM [%s]", self._context.vm.name)
+        output = ssh_client.run_command("update-waagent-conf Debug.EtpCollectionPeriod=30", use_sudo=True)
+        log.info("Updated waagent conf with Debug.ETPCollectionPeriod=30 completed:\n%s", output)
+
+        # Add CSE to the test VM twice to ensure its events directory still exists after re-enabling
+        log.info("")
+        log.info("Add CSE to the test VM...")
+        cse = VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.CustomScript, resource_name="CustomScript")
+        cse.enable(settings={'commandToExecute': "echo 'enable'"})
+        cse.assert_instance_view()
+
+        log.info("")
+        log.info("Add CSE to the test VM again...")
+        cse.enable(settings={'commandToExecute': "echo 'enable again'"})
+        cse.assert_instance_view()
+
+        # Check agent log to verify ETP is enabled
+        command = "agent_ext_workflow-check_data_in_agent_log.py --data 'Extension Telemetry pipeline enabled: True'"
+        log.info("")
+        log.info("Check agent log to verify ETP is enabled...")
+        log.info("Remote command [%s] completed:\n%s", command, ssh_client.run_command(command))
+
+        # Add good extension events for each extension and check that the TelemetryEventsCollector collects them
+        # TODO: Update test suite to check that the agent is picking up the events generated by the extension, instead
+        #  of generating on the extensions' behalf
+        #  (work item - https://dev.azure.com/msazure/One/_workitems/edit/24903999)
+        log.info("")
+        log.info("Add good extension events and check they are reported...")
+        max_events = random.randint(10, 50)
+        output = ssh_client.run_command(f"ext_telemetry_pipeline-add_extension_events.py "
```

Comment:
Recently Norberto wrapper for remote scripts run that would differentiate the assertions errors vs other errors. You may want to check that method. Your ssh run will raise as command error//unexpected error for assertions errors which confuses while debugging things

Reply:
Thanks for pointing this out - updated to use run_remote_test